apps/docs/content/guides/auth/social-login/auth-google.mdx

```diff
@@ -341,6 +341,83 @@ Most web apps and websites can utilize Google's [personalized sign-in buttons](h
    // Use 'nonce' when invoking the supabase.auth.signInWithIdToken() method
    ```
 
+### One-tap with NextJS
+
+If you're integrating Google One-Tap with your NextJS application, you can refer to the example below to get started:
+
+```tsx
+'use client'
+
+import Script from 'next/script'
+import { createClient } from '@/utils/supabase/client'
+import { CredentialResponse } from 'google-one-tap'
+import { useRouter } from 'next/navigation'
+import { useEffect } from 'react'
+
+const OneTapComponent = () => {
+  const router = useRouter()
+
+  // generate nonce to use for google id token sign-in
+  const generateNonce = async (): Promise<string[]> => {
+    const nonce = btoa(String.fromCharCode(...crypto.getRandomValues(new Uint8Array(32))))
+    const encoder = new TextEncoder()
+    const encodedNonce = encoder.encode(nonce)
+    const hashBuffer = await crypto.subtle.digest('SHA-256', encodedNonce)
+    const hashArray = Array.from(new Uint8Array(hashBuffer))
+    const hashedNonce = hashArray.map((b) => b.toString(16).padStart(2, '0')).join('')
+
+    return [nonce, hashedNonce]
+  }
+
+  useEffect(() => {
+    const initializeGoogleOneTap = () => {
+      window.onload = async () => {
+        const [nonce, hashedNonce] = await generateNonce()
+        console.log('Nonce: ', nonce, hashedNonce)
+        /* global google */
+        google.accounts.id.initialize({
+          client_id: process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID,
+          callback: async (response: CredentialResponse) => {
+            try {
+              const supabase = createClient()
+
+              // send id token returned in response.credential to supabase
+              const { data, error } = await supabase.auth.signInWithIdToken({
+                provider: 'google',
+                token: response.credential,
+                nonce,
+              })
+
+              if (error) throw error
+              console.log('Session data: ', data)
+              console.log('Successfully logged in with Google One Tap')
+
+              // redirect to protected page
+              router.push('/')
+            } catch (error) {
+              console.error('Error logging in with Google One Tap', error)
+            }
+          },
+          nonce: hashedNonce,
+          use_fedcm_for_prompt: true,
```

Comment:
This option was a little puzzling for me, can we consider adding a note adding an explanation or a reference to the section where [the use of the flag is described](https://supabase.com/docs/guides/auth/social-login/auth-google#google-pre-built)?

>To make your app compatible with Chome's third-party-cookie phase-out, make sure to set data-use_fedcm_for_prompt to true.